sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/handler/ReceiverUnsettledDeliveries.java

```diff
@@ -0,0 +1,730 @@
+        }
+
+        /**
+         * Check if the work has timed out.
+         *
+         * @return {@code true} if the work has timed out, {@code false} otherwise.
+         */
+        boolean hasTimedout() {
+            return expirationTime != null && expirationTime.isBefore(Instant.now());
+        }
+
+        /**
+         * Gets the {@link Mono} upon subscription starts the work by requesting ProtonJ library to send
+         * disposition frame to settle a delivery on the broker, and this Mono terminates once the broker
+         * acknowledges with disposition frame indicating settlement outcome (a.k.a. remote-outcome)
+         * The Mono can terminate if the configured timeout elapses or cannot initiate the request to
+         * ProtonJ library.
+         *
+         * @return the mono
+         */
+        Mono<Void> getMono() {
+            return mono;
+        }
+
+        /**
+         * Sets the {@link Mono}, where the application can obtain cached version of it
+         * from {@link DispositionWork#getMono()} and subscribe to start the work, the mono terminates
+         * upon the successful or unsuccessful completion of the work.
+         *
+         * @param mono the mono
+         */
+        void setMono(Mono<Void> mono) {
+            // cache() the mono to replay the result when subscribed more than once, avoid multiple
+            // disposition placement (and enables a possible second subscription to be safe when closing
+            // the UnsettledDeliveries type).
+            this.mono = mono.cache();
+        }
+
+        /**
+         * Check if this work is already completed.
+         *
+         * @return {@code true} if the work is completed, {@code true} otherwise.
+         */
+        boolean isCompleted() {
+            return this.get();
+        }
+
+        /**
+         * The function invoked once the application start the work by subscribing to the {@link Mono}
+         * obtained from {@link DispositionWork#getMono()}.
+         *
+         * @param monoSink the {@link MonoSink} to notify the completion of the work, which triggers
+         *                 termination of the same {@link Mono} that started the work.
+         */
+        void onStart(MonoSink<Void> monoSink) {
+            this.monoSink = monoSink;
+            expirationTime = Instant.now().plus(timeout);
+        }
+
+        /**
+         * The function invoked when the work is about to be restarted/retried. The broker may return an
+         * outcome named 'Rejected' if it is unable to attain the desired-outcome that the application
+         * specified in the disposition frame; in this case, the work is retried based on the configured
+         * retry settings.
+         *
+         * @param error the error that the broker returned upon Reject-ing the last work execution attempting
+         *              the disposition.
+         */
+        void onRetriableRejectedOutcome(Throwable error) {
```

Comment:
Actually, choice of this name i.e. with prefix 'Rejected' was intentional; the context was -

1. Today we know we have to only retry on "Rejected" outcomes out of the "known" seven standard state types represented by the enum DeliveryState.DeliveryStateType

2. In the future, if we want to support retry on any "user-defined" state (i.e., not in the "known" states but a state specialized to the broker), then the support won't come free, we might have to do some investigation like - the Ctr API for the client to pass retriable custom states / sendDisposition overload taking transient flag / maybe it's more involved. At the moment, we don't know or have visibility into how that story looks like.

This why we choose to _specialize_ the private method names as "handleRetriableRejectedRemoteOutcome", and companion "onRetriableRejectedOutcome" with JavaDoc, because we can't _generalize_ the retry concept yet without knowing the future requirement

Reply:
IOW, naming hints to the future dev working on any custom states that the current disposition retry may not be generic enough :blush: , so some investigation is required.

Reply:
Agree! Let's keep the name. If there is a new error in the future, then we decide at that time.